host/sysfs/gpio.go

```diff
@@ -365,9 +365,17 @@ func (p *Pin) haltEdge() error {
 		if err := seekWrite(p.fEdge, bNone); err != nil {
```

Comment:
In hindsight, I wonder if it's a good idea to reset the edge. Maybe we should have a separate function after all...

Basically, I'd like the following to work:
- In()
- WaitForEdge()
- Foo() (from another goroutine)
- WaitForEdge()

Where I had initially imagined that Foo() would be Halt(), but I'm less sure about it now.

Reply:
Not clear to me: 

```
In()
WaitForEdge()
go Foo()
WaitForEdge()
```

should work, while 

```
In()
WaitForEdge()
go func() {
  Foo()
  WaitForEdge()
}
```

wouldn't work, unless WaitForEdge is teansformed to a wrapper, calling `wait` internally ONLY ONCE.

Could be achieved with a channel for example: wait is only called if the channel already exists, all succesive calls to WaitForEdge try to read from the channel and when wait returns the channel is closed. This is the simplest approach I could imagine for a "fan out" or "NotifyAll".